lib/optparse.rb

```diff
@@ -461,7 +461,7 @@ def self.candidate(key, icase = false, pat = nil, &block)
       candidates
     end
 
-    def candidate(key, icase = false, pat = nil)
+    def candidate(key, icase = false, pat = nil, &)
```

Comment:
This is a syntax error in 3.0 or earlier, and unacceptable for optparse that still supports older versions.

```suggestion
    def candidate(key, icase = false, pat = nil, &_)
```